src/popup/router/pages/Retip.vue

```diff
@@ -31,71 +28,118 @@
 
 <script>
 import { mapGetters, mapState } from 'vuex';
-import BigNumber from 'bignumber.js';
-import tipping from 'aepp-raendom/src/utils/tippingContractUtil';
+import tipping from 'tipping-contract/util/tippingContractUtil';
 import { MAGNITUDE, calculateFee, TX_TYPES } from '../../utils/constants';
+import { convertToken } from '../../utils/helper';
 import deeplinkApi from '../../../mixins/deeplinkApi';
 import AmountSend from '../components/AmountSend';
 import UrlStatus from '../components/UrlStatus';
 import Button from '../components/Button';
+import BalanceInfo from '../components/BalanceInfo';
 
 export default {
   mixins: [deeplinkApi],
-  components: { AmountSend, UrlStatus, Button },
+  components: { AmountSend, UrlStatus, Button, BalanceInfo },
   data: () => ({
     tip: {},
     amount: '',
     loading: false,
   }),
   computed: {
     ...mapGetters(['tippingSupported']),
+    ...mapState('fungibleTokens', ['selectedToken']),
     ...mapState({
       tippingV1: 'tippingV1',
+      tippingV2: 'tippingV2',
       urlStatus(state, getters) {
         return getters['tipUrl/status'](this.tip.url);
       },
       isMinTipAmountError(state, { minTipAmount }) {
         return this.amount < minTipAmount;
       },
-      isAmountValid({ sdk, balance }, { account, activeNetwork }) {
-        if (!sdk) return false;
+      isAmountValid({ sdk, balance }, { account }) {
```

Comment:
This function has the overlapping checks with `errorMessage`, proposing to return error message from here instead.